lib/cog/repository/bundles.ex

```diff
@@ -631,6 +631,40 @@ defmodule Cog.Repository.Bundles do
     |> preload
   end
 
+  defp update_version!(version, params) do
+    params = Map.merge(params, Map.take(params["config_file"], ["author", "homepage"]))
+
+    version
+    |> BundleVersion.changeset(params)
+    |> Repo.update!
+    |> preload
+  end
+
+  defp new_or_updated_version!(bundle, params) do
+    # Check for an old version
+    version = params["config_file"]["version"]
+    result =
+      Repo.one(from bv in BundleVersion,
+               where: bv.bundle_id == ^bundle.id,
+               where: bv.version == ^version)
+
+    case result do
+      nil ->
+        # Create a new version if there isn't one
+        new_version!(bundle, params)
+      old_version ->
+        # If there is a version update it and then delete any commands and
+        # templates associated with it.
+        new_version = update_version!(old_version, params)
+        |> Repo.preload(:templates)
+
+        Enum.each(new_version.commands, &Repo.delete!/1)
+        Enum.each(new_version.templates, &Repo.delete!/1)
```

Comment:
The old commands and templates are removed here and they are reinstalled later in the transaction. We could go through and update all of the associations instead of just reinstalling, but that is a lot more logic and seemed more likely to introduce bugs. With this method bundle installation continues just as it did normally with the exception that we reference the old bundle version instead of a newly created one.